apps/files/src/components/Collaborators/CollaboratorsEditOptions.vue

```diff
@@ -0,0 +1,88 @@
+<template>
+  <oc-grid gutter="small" childWidth="1-1">
+    <roles-select
+      :selectedRole="role"
+      @roleSelected="selectRole"
+    />
+    <additional-permissions
+      :selectedRole="role"
+      :collaborator="collaborator"
+      @permissionChecked="checkAdditionalPermissions"
+    />
+  </oc-grid>
+</template>
+
+<script>
+import collaboratorsMixins from '../../mixins/collaborators'
+
+const RolesSelect = () => import('./RolesSelect.vue')
+const AdditionalPermissions = () => import('./AdditionalPermissions.vue')
+
+export default {
+  name: 'CollaboratorsEditOptions',
+  mixins: [
+    collaboratorsMixins
+  ],
+  props: {
+    existingRole: {
+      type: Object,
+      required: false
+    },
+    collaborator: {
```

Comment:
let's see if we can get rid of this?

and just pass in additional permissions as a property for better encapsulation

so these components and also AdditionalPermissions.vue don't need to know about the collaborator